### PR TITLE
[enhancement] speed up week_of_year by pre_calc table

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
select  count(LO_ORDERDATE)  FROM  lineorder  WHERE  weekofyear(LO_ORDERDATE)  =  6;
before: 19 sec
after: 13 sec
we make a static const uint8_t array week_of_year_table in VecDateTimeValue.
if the year is between [1950, 2029], we get the result from array week_of_year_table, otherwise calculate the result as before.

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Gabriel39 — COMMENTED** [review]
> before: 13 sec
> after: 19 sec

make slower?

**Comment by kpfly** [comment]
Why does it have a negative impact on performance ？

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by englefly** [comment]
> Why does it have a negative impact on performance ？



> Why does it have a negative impact on performance ？



> 



> > before: 13 sec
> > after: 19 sec
> 
> make slower?



> > before: 13 sec
> > after: 19 sec
> 
> make slower?

sorry，笔误

File: be/src/vec/runtime/vdatetime_value.cpp [hunk 1]
```diff
@@ -837,6 +837,9 @@ bool VecDateTimeValue::to_format_string(const char* format, int len, char* to) c
 }
 
 uint8_t VecDateTimeValue::calc_week(const VecDateTimeValue& value, uint8_t mode, uint32_t* year) {
+    if (value._year >= 1949 && value._year < 2030) {
+        return week_of_year_table[value._year - 1950][value._month][value._day];
```
Comment (Gabriel39) [inline be/src/vec/runtime/vdatetime_value.cpp:841]: `mode` also needs to be considered

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

File: be/src/vec/runtime/vdatetime_value.cpp (hunk 1 above)
Comment (englefly) [reply be/src/vec/runtime/vdatetime_value.cpp:841]: in doris，mode is a hard-coded value 3.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**